airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -409,23 +409,22 @@ def read_records(
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
+        yield from self._read_pages(
+            lambda req, res, state, _slice: self.parse_response(res, stream_slice=_slice, stream_state=state), stream_slice, stream_state
+        )
+
+    def _read_pages(
```

Comment:
what's the return type of this method/the input record generator fn?

Reply:
added type hints